### ISSUE TITLE
Can't open in chrome unless it's in Incognito page

### ISSUE THREAD

**Issue by hugalafutro** [jlesage/docker-jdownloader-2#90]
Hi,
I have a strange issue with all your images using novnc, maybe you'll have some idea why I can only open your apps in Incognito mode?

edit: looking at the logs the only difference I see is when it works it says `webSocketsHandshake: using binary/raw encoding` after the ws handshake, but when it fails it says `webSocketsHandshake: using base64 encoding`. Is there any way I can make it use the one it uses in incognito in normal page?

I am using a chrome clone Vivaldi. I tried disabling all extensions to no avail. I thought it's the adblocker, but that is the only extension enabled in incognito where the novnc already works. 

When opening in normal page I see error about websocket `WebSocket connection to 'ws://192.168.1.249:5850/websockify' failed: ` in web console:
```
DevTools failed to load source map: Could not load content for http://192.168.1.249:5850/css/bootstrap.min.css.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
ui.js:62 Starting UI...
DevTools failed to load source map: Could not load content for chrome-extension://mmhlniccooihdimnnjhamobppdhaolme/lib/tp/browser-polyfill.min.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
util.js:218  Msg: noVNC ready: native WebSockets, canvas rendering
Util.Warn @ util.js:218
_updateState @ rfb.js:462
RFB @ rfb.js:222
initRFB @ ui.js:151
init @ ui.js:94
start @ ui.js:63
(anonymous) @ (index):63
j @ jquery.min.js?v=b80cd69e7d:2
k @ jquery.min.js?v=b80cd69e7d:2
setTimeout (async)
(anonymous) @ jquery.min.js?v=b80cd69e7d:2
i @ jquery.min.js?v=b80cd69e7d:2
fireWith @ jquery.min.js?v=b80cd69e7d:2
fire @ jquery.min.js?v=b80cd69e7d:2
i @ jquery.min.js?v=b80cd69e7d:2
fireWith @ jquery.min.js?v=b80cd69e7d:2
ready @ jquery.min.js?v=b80cd69e7d:2
S @ jquery.min.js?v=b80cd69e7d:3
util.js:218 
Util.Warn @ util.js:218
_updateState @ rfb.js:462
connect @ rfb.js:243
start @ ui.js:65
(anonymous) @ (index):63
j @ jquery.min.js?v=b80cd69e7d:2
k @ jquery.min.js?v=b80cd69e7d:2
setTimeout (async)
(anonymous) @ jquery.min.js?v=b80cd69e7d:2
i @ jquery.min.js?v=b80cd69e7d:2
fireWith @ jquery.min.js?v=b80cd69e7d:2
fire @ jquery.min.js?v=b80cd69e7d:2
i @ jquery.min.js?v=b80cd69e7d:2
fireWith @ jquery.min.js?v=b80cd69e7d:2
ready @ jquery.min.js?v=b80cd69e7d:2
S @ jquery.min.js?v=b80cd69e7d:3
android-chrome-192x192.png:1 GET http://192.168.1.249:5850/images/icons/images/icons/android-chrome-192x192.png?v=d46d703046 404 (Not Found)
192.168.1.249/:1 Error while trying to use the following icon from the Manifest: http://192.168.1.249:5850/images/icons/images/icons/android-chrome-192x192.png?v=d46d703046 (Download error or resource isn't a valid image)
websock.js:292 WebSocket connection to 'ws://192.168.1.249:5850/websockify' failed: 
open @ websock.js:292
_connect @ rfb.js:346
_updateState @ rfb.js:488
connect @ rfb.js:243
start @ ui.js:65
(anonymous) @ (index):63
j @ jquery.min.js?v=b80cd69e7d:2
k @ jquery.min.js?v=b80cd69e7d:2
util.js:218 WebSocket on-error event
Util.Warn @ util.js:218
(anonymous) @ rfb.js:214
(anonymous) @ websock.js:325
error (async)
open @ websock.js:323
_connect @ rfb.js:346
_updateState @ rfb.js:488
connect @ rfb.js:243
start @ ui.js:65
(anonymous) @ (index):63
j @ jquery.min.js?v=b80cd69e7d:2
k @ jquery.min.js?v=b80cd69e7d:2
setTimeout (async)
(anonymous) @ jquery.min.js?v=b80cd69e7d:2
i @ jquery.min.js?v=b80cd69e7d:2
fireWith @ jquery.min.js?v=b80cd69e7d:2
fire @ jquery.min.js?v=b80cd69e7d:2
i @ jquery.min.js?v=b80cd69e7d:2
fireWith @ jquery.min.js?v=b80cd69e7d:2
ready @ jquery.min.js?v=b80cd69e7d:2
S @ jquery.min.js?v=b80cd69e7d:3
util.js:218 WebSocket on-close event
Util.Warn @ util.js:218
(anonymous) @ rfb.js:193
(anonymous) @ websock.js:320
util.js:220  Msg: Server disconnected (code: 1006)
Util.Error @ util.js:220
_updateState @ rfb.js:460
_fail @ rfb.js:530
(anonymous) @ rfb.js:209
(anonymous) @ websock.js:320
util.js:218 
Util.Warn @ util.js:218
_updateState @ rfb.js:462
(anonymous) @ rfb.js:513
setTimeout (async)
_updateState @ rfb.js:512
_fail @ rfb.js:530
(anonymous) @ rfb.js:209
(anonymous) @ websock.js:320
```

docker logs when opening from normal page where it fails:
```
03/01/2022 11:04:18 Got connection from client 127.0.0.1
03/01/2022 11:04:18   other clients:
03/01/2022 11:04:18 Got 'ws' WebSockets handshake
03/01/2022 11:04:18   - webSocketsHandshake: using base64 encoding
03/01/2022 11:04:18   - WebSockets client version hybi-13
caught signal: 11
03/01/2022 11:04:18 deleted 60 tile_row polling images.
[x11vnc] starting...
03/01/2022 11:04:18 passing arg to libvncserver: -rfbport
03/01/2022 11:04:18 passing arg to libvncserver: 5900
03/01/2022 11:04:18 passing arg to libvncserver: -rfbportv6
03/01/2022 11:04:18 passing arg to libvncserver: -1
03/01/2022 11:04:18 passing arg to libvncserver: -httpportv6
03/01/2022 11:04:18 passing arg to libvncserver: -1
03/01/2022 11:04:18 passing arg to libvncserver: -desktop
03/01/2022 11:04:18 passing arg to libvncserver: JDownloader 2
03/01/2022 11:04:18 x11vnc version: 0.9.14 lastmod: 2015-11-14  pid: 193365
03/01/2022 11:04:18 Using X display :0
03/01/2022 11:04:18 rootwin: 0x43 reswin: 0x200001 dpy: 0xd2ff8b00
03/01/2022 11:04:18
03/01/2022 11:04:18 ------------------ USEFUL INFORMATION ------------------
03/01/2022 11:04:18 X DAMAGE available on display, using it for polling hints.
03/01/2022 11:04:18   To disable this behavior use: '-noxdamage'
03/01/2022 11:04:18
03/01/2022 11:04:18   Most compositing window managers like 'compiz' or 'beryl'
03/01/2022 11:04:18   cause X DAMAGE to fail, and so you may not see any screen
03/01/2022 11:04:18   updates via VNC.  Either disable 'compiz' (recommended) or
03/01/2022 11:04:18   supply the x11vnc '-noxdamage' command line option.
03/01/2022 11:04:18 X COMPOSITE available on display, using it for window polling.
03/01/2022 11:04:18   To disable this behavior use: '-noxcomposite'
03/01/2022 11:04:18
03/01/2022 11:04:18 Wireframing: -wireframe mode is in effect for window moves.
03/01/2022 11:04:18   If this yields undesired behavior (poor response, painting
03/01/2022 11:04:18   errors, etc) it may be disabled:
03/01/2022 11:04:18    - use '-nowf' to disable wireframing completely.
03/01/2022 11:04:18    - use '-nowcr' to disable the Copy Rectangle after the
03/01/2022 11:04:18      moved window is released in the new position.
03/01/2022 11:04:18   Also see the -help entry for tuning parameters.
03/01/2022 11:04:18   You can press 3 Alt_L's (Left "Alt" key) in a row to
03/01/2022 11:04:18   repaint the screen, also see the -fixscreen option for
03/01/2022 11:04:18   periodic repaints.
03/01/2022 11:04:18 GrabServer control via XTEST.
03/01/2022 11:04:18
03/01/2022 11:04:18 Scroll Detection: -scrollcopyrect mode is in effect to
03/01/2022 11:04:18   use RECORD extension to try to detect scrolling windows
03/01/2022 11:04:18   (induced by either user keystroke or mouse input).
03/01/2022 11:04:18   If this yields undesired behavior (poor response, painting
03/01/2022 11:04:18   errors, etc) it may be disabled via: '-noscr'
03/01/2022 11:04:18   Also see the -help entry for tuning parameters.
03/01/2022 11:04:18   You can press 3 Alt_L's (Left "Alt" key) in a row to
03/01/2022 11:04:18   repaint the screen, also see the -fixscreen option for
03/01/2022 11:04:18   periodic repaints.
03/01/2022 11:04:18
03/01/2022 11:04:18 XKEYBOARD: number of keysyms per keycode 7 is greater
03/01/2022 11:04:18   than 4 and 51 keysyms are mapped above 4.
03/01/2022 11:04:18   Automatically switching to -xkb mode.
03/01/2022 11:04:18   If this makes the key mapping worse you can
03/01/2022 11:04:18   disable it with the "-noxkb" option.
03/01/2022 11:04:18   Also, remember "-remap DEAD" for accenting characters.
03/01/2022 11:04:18
03/01/2022 11:04:18 X FBPM extension not supported.
Xlib:  extension "DPMS" missing on display ":0".
03/01/2022 11:04:18 X display is not capable of DPMS.
03/01/2022 11:04:18 --------------------------------------------------------
03/01/2022 11:04:18
03/01/2022 11:04:18 Default visual ID: 0x21
03/01/2022 11:04:18 Read initial data from X display into framebuffer.
03/01/2022 11:04:18 initialize_screen: fb_depth/fb_bpp/fb_Bpl 24/32/7680
03/01/2022 11:04:18
03/01/2022 11:04:18 X display :0 is 32bpp depth=24 true color
03/01/2022 11:04:18
03/01/2022 11:04:18 Listening for VNC connections on TCP port 5900
03/01/2022 11:04:18
03/01/2022 11:04:18 Xinerama is present and active (e.g. multi-head).
03/01/2022 11:04:18 Xinerama: number of sub-screens: 1
03/01/2022 11:04:18 Xinerama: no blackouts needed (only one sub-screen)
03/01/2022 11:04:18
03/01/2022 11:04:18 fb read rate: 1461 MB/sec
03/01/2022 11:04:18 fast read: reset -wait  ms to: 10
03/01/2022 11:04:18 fast read: reset -defer ms to: 10
03/01/2022 11:04:18 The X server says there are 10 mouse buttons.
03/01/2022 11:04:18 screen setup finished.
03/01/2022 11:04:18

The VNC desktop is:      43ed0e4e3f6e:0
PORT=5900

******************************************************************************
Have you tried the x11vnc '-ncache' VNC client-side pixel caching feature yet?

The scheme stores pixel data offscreen on the VNC viewer side for faster
retrieval.  It should work with any VNC viewer.  Try it by running:

    x11vnc -ncache 10 ...

One can also add -ncache_cr for smooth 'copyrect' window motion.
More info: http://www.karlrunge.com/x11vnc/faq.html#faq-client-caching
```

docker logs from opening in incognito where it works:
```
03/01/2022 11:06:26 Got connection from client 127.0.0.1
03/01/2022 11:06:26   other clients:
03/01/2022 11:06:26 Got 'ws' WebSockets handshake
03/01/2022 11:06:26 Got protocol: binary
03/01/2022 11:06:26   - webSocketsHandshake: using binary/raw encoding
03/01/2022 11:06:26   - WebSockets client version hybi-13
03/01/2022 11:06:26 Disabled X server key autorepeat.
03/01/2022 11:06:26   to force back on run: 'xset r on' (3 times)
03/01/2022 11:06:26 incr accepted_client=1 for 127.0.0.1:39258  sock=10
03/01/2022 11:06:26 Client Protocol Version 3.8
03/01/2022 11:06:26 Protocol version sent 3.8, using 3.8
03/01/2022 11:06:26 rfbProcessClientSecurityType: executing handler for type 1
03/01/2022 11:06:26 rfbProcessClientSecurityType: returning securityResult for client rfb version >= 3.8
03/01/2022 11:06:26 Pixel format for client 127.0.0.1:
03/01/2022 11:06:26   32 bpp, depth 24, little endian
03/01/2022 11:06:26   true colour: max r 255 g 255 b 255, shift r 16 g 8 b 0
03/01/2022 11:06:26 no translation needed
03/01/2022 11:06:26 Enabling NewFBSize protocol extension for client 127.0.0.1
03/01/2022 11:06:26 Enabling full-color cursor updates for client 127.0.0.1
03/01/2022 11:06:26 Using image quality level 6 for client 127.0.0.1
03/01/2022 11:06:26 Using JPEG subsampling 0, Q79 for client 127.0.0.1
03/01/2022 11:06:26 Using compression level 9 for client 127.0.0.1
03/01/2022 11:06:26 Enabling LastRect protocol extension for client 127.0.0.1
03/01/2022 11:06:26 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFECC)
03/01/2022 11:06:26 Using tight encoding for client 127.0.0.1
03/01/2022 11:06:27 client_set_net: 127.0.0.1  0.0000
03/01/2022 11:06:27 created   xdamage object: 0x200040
03/01/2022 11:06:35 created selwin: 0x200041
03/01/2022 11:06:35 called initialize_xfixes()
```

Opening in incognito it works as expected. I don;t blame your work it's obviously the browser as it works elsewhere, but not in my browser of choice, so I thought you might see what's going on perhaps. It's just a minor hassle when I open the page every time, then go 'Doh and re-open it in incognito.

Happy new year and thanks for any insights!

**Comment by jlesage** [comment]
Did you try to clear browser's cache ?  This usually fix this issue.

**Comment by hugalafutro** [comment]
Hi I have indeed try that numerous times to no avail, but today I found it has to do something with my user profile. I deleted my old profile, created new, synced it with data from server to get my bookmarks+extensions back and now it works, I just have to log in to million pages again now.

tl;dr; Resetting user profile fixed the issue.
